schemas/layer-4.cue

```diff
@@ -1,37 +1,108 @@
+#Assessment: {
+	// RequirementID uniquely identifies the requirement being tested
+	"requirement-id": string @go(RequirementId)
+	// Applicability provides identifier strings to determine when this assessment is applicable
+	applicability: [...string]
+	// Description provides a detailed explanation of the assessment
+	description: string
+	// Result communicates whether the assessment has been run, and if so, the outcome(s)
+	result: #Result
+	// Message describes the result of the assessment
+	message: string
+	// Procedures defines the assessment procedures associated with the assessment
+	procedures: [...#AssessmentProcedure]
+	// RunDuration is the time it took to run the assessment
+	"run-duration"?: string @go(RunDuration)
+	// Value is the object that was returned during the assessment
+	value?: _
+	// Changes describes changes that were made during the assessment
+	changes?: [string]: #Change
+}
+
+// AssessmentProcedure describes a testing procedure for evaluating a Layer 2 control requirement.
+#AssessmentProcedure: {
+	// Id uniquely identifies the assessment procedure being executed
+	id: string
+	// Name provides a summary of the procedure
+	name: string
+	// Description provides a detailed explanation of the procedure
+	description: string
+	// Method describe the high-level method used to determine the results of the procedure
+	method: #ProcedureMethod
+	// Run is a boolean indicating whether the procedure was run or not. When run is true, result is expected to be present
+	run: bool
+	// RemediationGuide provides a URL with remediation guidance associated with the control's assessment requirement and this specific assessment procedure
+	"remediation-guide"?: #URL @go(RemediationGuide)
+	// Documentation provides a URL to documentation that describes how the assessment procedure evaluates the control requirement
+	documentation?: #URL
+	// Steps provides the address for the assessment steps executed
+	"steps"?: [...string]
 }
 
-#AssessmentResult: {
-    result: #Result
-    name: string
-    description: string
-    message: string
-    "function-address": string
-    change?: #Change
-    value?: _
+// Additional constraints on Assessment Procedure.
+#AssessmentProcedure: {
+	run: false
+	// Message describes the result of the procedure
+	message?: string
+	// Result communicates the outcome(s) of the procedure
+	result?: ("Not Run" | *null) @go(Result,optional=nillable)
+} | {
+	run:     true
+	message!: string
+	result!: #ResultWhenRun
 }
 
-#Result: "Passed" | "Failed" | "Needs Review"
+// Result describes valid assessment outcomes before and after execution.
+#Result: #ResultWhenRun | "Not Run"
 
+// Result describes the outcome(s) of an assessment procedure when it is executed.
+#ResultWhenRun: "Passed" | "Failed" | "Needs Review" | "Not Applicable" | "Unknown"
+
+// ProcedureMethod describes method options that can be used to determine the results
+#ProcedureMethod: "Test" | "Observation"
```

Comment:
I think we need to clarify the language here. In my experience testing involves observation.

What value does this provide to the intended data consumers? How do you see them using this fact?